### PR TITLE
securitycenter: update golang sample with new mutable field

### PR DESCRIPTION
**Pull request by hannah-tsai**
Fixes b/155346619

Update golang sample with new mutable field for UpdateNotificationConfig endpoint: filter

### CODE REVIEW

File: securitycenter/notifications/update_notification_config.go [hunk 1]
```diff
@@ -39,14 +39,16 @@ func updateNotificationConfig(w io.Writer, orgID string, notificationConfigID st
 	defer client.Close()
 
 	updatedDescription := "Updated sample config"
+	updatedFilter := "state = \"INACTIVE\""
```
Comment (tbpg) [inline securitycenter/notifications/update_notification_config.go:42]:
```suggestion
	updatedFilter := `state = "INACTIVE"`
```

File: securitycenter/notifications/update_notification_config.go [hunk 2]
```diff
@@ -39,14 +39,16 @@ func updateNotificationConfig(w io.Writer, orgID string, notificationConfigID st
...
 	req := &securitycenterpb.UpdateNotificationConfigRequest{
 		NotificationConfig: &securitycenterpb.NotificationConfig{
 			Name:        fmt.Sprintf("organizations/%s/notificationConfigs/%s", orgID, notificationConfigID),
 			Description: updatedDescription,
 			PubsubTopic: updatedPubsubTopic,
+			StreamingConfig: &securitycenterpb.StreamingConfig{Filter: updatedFilter}
```
Comment (tbpg) [inline securitycenter/notifications/update_notification_config.go:48]:
```suggestion
			StreamingConfig: &securitycenterpb.StreamingConfig{Filter: updatedFilter},
```

Comment (hannah-tsai) [reply securitycenter/notifications/update_notification_config.go]: fixed

File: securitycenter/notifications/update_notification_config.go (hunk 1 above)
Comment (hannah-tsai) [reply securitycenter/notifications/update_notification_config.go]: fixed!

**Comment by hannah-tsai** [comment]
Hi, not sure why the some test failed after format fix, could you help to restart the tests?

File: securitycenter/notifications/update_notification_config.go [hunk 3]
```diff
@@ -39,14 +39,16 @@ func updateNotificationConfig(w io.Writer, orgID string, notificationConfigID st
 	defer client.Close()
 
 	updatedDescription := "Updated sample config"
+	updatedFilter := `state = \"INACTIVE\"`
```
Comment (tbpg) [inline securitycenter/notifications/update_notification_config.go:42]:
```suggestion
	updatedFilter := `state = "INACTIVE"`
```

File: securitycenter/notifications/update_notification_config.go [hunk 4]
```diff
@@ -39,14 +39,16 @@ func updateNotificationConfig(w io.Writer, orgID string, notificationConfigID st
...
 	req := &securitycenterpb.UpdateNotificationConfigRequest{
 		NotificationConfig: &securitycenterpb.NotificationConfig{
 			Name:        fmt.Sprintf("organizations/%s/notificationConfigs/%s", orgID, notificationConfigID),
 			Description: updatedDescription,
 			PubsubTopic: updatedPubsubTopic,
+			StreamingConfig: &securitycenterpb.StreamingConfig{Filter: updatedFilter},
```
Comment (tbpg) [inline securitycenter/notifications/update_notification_config.go:48]: Looks like this field isn't in the generated `NotificationConfig` type yet. When was it released? We regenerate the client every day. That's currently causing a compilation failure for this change. So, we won't be able to merge this until the generated type is updated.

See https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/securitycenter/v1?tab=doc#NotificationConfig.

**Comment by tbpg** [comment]
Also, looks like you need to run `gofmt -w .` to update the formatting.

File: securitycenter/notifications/update_notification_config.go (hunk 3 above)
Comment (hannah-tsai) [reply securitycenter/notifications/update_notification_config.go]: resolved.

**Comment by tbpg** [comment]
Unfortunately, it looks like the field still hasn't been generated. When was it released? Is it available in other languages?

cc @codyoss

**Comment by codyoss** [comment]
Looks like the field name and type is incorrect? 

https://github.com/googleapis/go-genproto/blob/1244ee217b7ef50797c0e4206d3b4b641901d48a/googleapis/cloud/securitycenter/v1/notification_config.pb.go#L70

https://github.com/googleapis/go-genproto/blob/1244ee217b7ef50797c0e4206d3b4b641901d48a/googleapis/cloud/securitycenter/v1/notification_config.pb.go#L160-L187

**Comment by hannah-tsai** [comment]
> Looks like the field name and type is incorrect?
> 
> https://github.com/googleapis/go-genproto/blob/1244ee217b7ef50797c0e4206d3b4b641901d48a/googleapis/cloud/securitycenter/v1/notification_config.pb.go#L70
> 
> https://github.com/googleapis/go-genproto/blob/1244ee217b7ef50797c0e4206d3b4b641901d48a/googleapis/cloud/securitycenter/v1/notification_config.pb.go#L160-L187

Thanks for pointing this out, I corrected the type but the Kokoro tests always seems to complete with one failed child job. Should I be concerned? https://source.cloud.google.com/results/invocations/665cd143-0bc8-4316-9e10-45bfb6d00629/targets/cloud-devrel%2Fgo%2Fgolang-samples%2Fpresubmit/log

**Pull request merged**